spec/support/shared_entity_specs.rb

```diff
@@ -70,3 +70,19 @@ def validate_values(value, parsed_value, type)
     end
   end
 end
+
+shared_examples "a relayable Entity" do
+  let(:instance) { described_class.new(data) }
+
+  context "signatures generation" do
+    it "computes correct signatures for the entity" do
+      xml = DiasporaFederation::Salmon::XmlPayload.pack(instance)
+      hash = DiasporaFederation::Salmon::XmlPayload.hash(xml.at_xpath("post/*[1]"))
```

Comment:
That's my fault, I had to verify it against ```hash[:author_signature]``` but not ```instance.author_signature```. The tests are written so the correct value is presented in instance constructor, but IRL it could be different (because the caller is unsure about XML structure which we consider to be the base for signature computation/verification).